website/docs/feature-flag-tutorials/java/spring-boot-implementing-feature-flags.md

```diff
@@ -0,0 +1,384 @@
+
+In this tutorial, you will learn how to use a feature flag in a Java Spring Boot app. We will use the [Spring Pet Clinic app](https://github.com/spring-projects/spring-petclinic) and the [Unleash Spring Boot SDK](https://github.com/Unleash/unleash-spring-boot-starter) to control how a [Spring Bean](https://docs.spring.io/spring-framework/reference/core/beans/definition.html) is used for a new page. 
+
+Here are the steps we will cover in this tutorial:
+1. [Feature flag best practices for back-end applications](#1-feature-flag-best-practices-for-server-side-apps)
+2. [Spin up a local flag provider](#2-install-a-local-feature-flag-provider)
+3. [Configure a feature flag](#3-create-and-configure-the-feature-flag)
+4. [Add Unleash to a Spring Boot app](#4-add-unleash-to-a-spring-boot-app)
+5. [Configure Spring Beans in your Spring Boot app](#5-configure-spring-beans-in-your-app)
+6. [Verify the toggle experience](#5-configure-spring-beans-in-your-app)
+
+
+## Prerequisites
+
+
+In this tutorial, you will need the following:
+- A web browser like Chrome or Firefox
+- Git
+- Docker
+- Maven or Gradle for building Java
+- (Optional) a code editor like IntelliJ IDEA
+
+![Java Spring Boot Architectural Diagram](/img/spring-boot-tutorial-architectural-diagram.png)
+
+This architecture diagram breaks down how the Java Spring Boot app works with Unleash to use feature flags.
+
+The Unleash Server is a Feature Flag Control Service for managing and storing your feature flags. It enables the retrieval of flag data and, particularly when not utilizing a user interface, supports sending data to and from the service. The Unleash Server has a UI for creating and managing projects and feature flags. API commands are also [available](/reference/api/unleash) to perform the same actions from your CLI or server-side app.
+
+
+## 1. Feature flag best practices for server-side apps
+
+
+Since Spring Boot is a framework used for Java backend apps, there are special security considerations to plan around when implementing feature flags.
+
+Most importantly, you must:
+- Limit feature flag payloads for scalability, security, and efficiency
+- Improve architectural resiliency with graceful degradation
+
+As your application scales, performance and resiliency become more critical and costly if not addressed. A feature flagging system should not be why your app slows down or fails. That’s why we recommend you account for this by reducing the size of your feature flag payloads. For example, instead of making one large call to retrieve flag statuses for all users as part of your configuration, group your users by specific attributes as part of your targeting rules that would be most relevant to your application.
+
+Additionally, you can cache your feature flag configuration to help reduce network round trips and dependency on external services. You can rely on the cache if your Feature Flag Control Service is unavailable, mitigating potential application failure.
```

Comment:
I would also remove the previous 2 paragraphs there

Reply:
We've been using this text for our tutorials to provide more perspective beyond just using a feature flag. It is meant to give devs insight into using feature flags at scale, in real world more complex environments so they're architecting it using some of our principles from our best practices documentation.